tests/integration/security/sds_ingress/util/util.go

```diff
@@ -644,3 +644,34 @@ func SetInstances(apps echo.Services) error {
 	}
 	return nil
 }
+
+// Some cloud platform may throw the following error during creation of the service with mixed TCP/UDP protocols:
+// "Error syncing load balancer: failed to ensure load balancer: mixed protocol is not supported for LoadBalancer".
+// Make sure the service is up and running before proceeding with the test.
+func WaitForIngressQUICService(t framework.TestContext, ns string) error {
+	return retry.UntilSuccess(func() error {
+		services, err := t.Clusters().Default().Kube().CoreV1().Services(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(services.Items) == 0 {
+			return fmt.Errorf("still waiting for the service in namespace %s to be created", ns)
+		}
+
+		// Fetch events to check for the service status
+		events, err := t.Clusters().Default().Kube().CoreV1().Events(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		// Verify that services are not stuck with creation error
+		for _, s := range services.Items {
```

Comment:
Would it be easier to check for a successful running event rather than a specific failure? Right now if the service failed with a different error message we wouldn't retry.

Reply:
Hello @jaellio ,

I would agree about it, but since the service is stuck in a "pending" state forever, I could not rely on this one, since it could be a matter of latency and I don't get any specific state at the end. It just a test timeout.
I could identify the issue only by checking the service event and make sure that it's not supported and not any latency issue.

Reply:
If the error is "mixed protocol is not supported for LoadBalancer" do we need to keep retrying?

Reply:
If the error is "mixed protocol is not supported for LoadBalancer" do we need to keep retrying?

Reply:
@jaellio 

No.
That's why when the error is identified, the function returns with an error.

I'm looping over services, because there are more that one service exists.
Then, I'm looping over the existing events to make sure I'm checking all events.

Once, the required even is found, I'm returning an error.

Reply:
Hi @jaellio 

Please, fix me if I'm wrong, but I'm not sure, the `UntilComplete` here is the best approach.
At the end, the function checks for the service events, and if condition event is not found, it should return nil and continue to run.

When using `UntilSuccess`, the function returns based on the success or failure state.
And that's what I would like to achieve - return nil if the service is running or return error if specific even found.

When using `UntilComplete`, the function does not stop execution based on success or failure, but keeps running until the timeout reached.

Reply:
The current approach with `UntilSuccess` create unnecessary retries. We don't need to keep retrying when you find the desired error state. By returning an error in `UntilSuccess` it will keep retrying which is unnecessary in this case. `UntilSuccess` will only stop retrying a function when the function returns nil or timesout. When we get the unsupported error we don't need to keep retrying, but will since the function is returning an error.

Reply:
Hi @jaellio 

I've modified the function to use the `UntilComplete` flow.